bessctl/measurement_utils.py

```diff
@@ -0,0 +1,393 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+# list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# * Neither the names of the copyright holders nor the names of their
+# contributors may be used to endorse or promote products derived from this
+# software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import print_function
+
+import copy
+import os
+import sys
+import time
+
+try:
+    this_dir = os.path.dirname(os.path.realpath(__file__))
+    bessctl = os.path.join(this_dir, 'bessctl')
+    sys.path.insert(1, os.path.join(this_dir, '../../../'))
+    from pybess.bess import *
+except ImportError:
+    print('Cannot import the API module (pybess)', file=sys.stderr)
+    raise
+
+
+def get_local_bess_handle():
+    bess = BESS()
+    try:
+        bess.connect()
+    except BESS.APIError:
+        raise Exception('BESS is not running')
+    return bess
+
+
+class PortStats(object):
+
+    """
+    Stores PMDPort stats and optional Measure module output and exports some
+    acceessors for convenience.
+    """
+
+    def __init__(self, throughput, rtt, jitter):
+        self.throughput = throughput
+        self.rtt = rtt
+        self.jitter = jitter
+
+    def pkts_in(self):
+        return self.throughput.inc.packets
+
+    def pkts_out(self):
+        return self.throughput.out.packets
+
+    def drops_in(self):
+        return self.throughput.inc.dropped
+
+    def drops_out(self):
+        return self.throughput.out.dropped
+
+    def bytes_in(self):
+        return self.throughput.inc.bytes
+
+    def bytes_out(self):
+        return self.throughput.out.bytes
+
+    def rtt(self, percentile=None):
+        """
+        Returns the dictionary of all RTT value if `percentile` is none.
+        Otherwise returns the `percentile`th percentile RTT.
+        """
+        if self.rtt is None:
+            return None
+        if percentile is None:
+            return self.rtt
```

Comment:
Nothing wrong with these as written, but it occurs to me that you could use:

    if self.rtt is None or percentile is None:
        return self.rtt

to save a couple of lines.  Same goes for `jitter` below.

Reply:
Definitely cleaner. Switched